controllers/helmchart_controller.go

```diff
@@ -493,9 +494,23 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
+	version, err := semver.NewVersion(helmChart.Metadata.Version)
+	if err != nil {
+		err = fmt.Errorf("load chart error: %w", err)
+		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+	}
+	// Replace / with - in version metadata for SemVer spec
+	version.SetMetadata(strings.Replace(artifact.Revision, "/", "-", -1))
```

Comment:
Sets the metadata as `main-<sha>` if a branch by replacing `/` with `-` as per spec.

Decided against `flux` metadata prefix for the time-being.